lib/Map/LeafletVisualizer.js

```diff
@@ -731,6 +731,26 @@ LeafletVisualizer.prototype.visualizersCallback = function(leafletScene, dataSou
     return [new LeafletGeomVisualizer(leafletScene, entities)];
 };
 
+/**
+ * Because we're on an old crappy version of Leaflet the click handling is a bit inconsistent - clicks for some vectors
+ * (e.g. paths, and CircleMarkers which are extensions of paths) propagate through to the map, whereas clicks for others
+ * (normal markers, i.e. any billboard) have their propagation stopped, which means that the clicks won't fall through
+ * to the base layers and feature picking will only work for the topmost vector. This hacks around it by manually
+ * refiring the click event on the base map after the feature is clicked.
+ *
+ * This still won't fix multiple vector layers on top of each other :(.
+ *
+ * TODO: Can remove this when we upgrade leaflet, as later versions have a completely overhauled event bubbling system.
+ *
+ */
+function featureClickedReraiseClick(visualizer, entity, e) {
+    if (visualizer._leafletScene.map._fireMouseEvent) {
+        visualizer._leafletScene.map._fireMouseEvent(e.originalEvent);
```

Comment:
Oh, nevermind, I see... You're only using this handler for the feature types known to have the problem.  Nice.